### PR TITLE
[gcp][feat] Add Vertex AI collection

### PR DESCRIPTION
**Pull request by 1101-1**
# Description

# To-Dos

- [x] I have created tests for any new or updated functionality.
- [x] I ran `tox` successfully.

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://inventory.fix.security/code-of-conduct).

### CODE REVIEW

File: plugins/gcp/fix_plugin_gcp/resources/aiplatform.py [hunk 1]
```diff
@@ -0,0 +1,3376 @@
...
+    def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
+        if tuned_model := self.tuned_model:
+            if endpoint := tuned_model.endpoint:
+                builder.add_edge(self, reverse=True, clazz=GcpVertexAIEndpoint, name=endpoint)
+
+
+resources: List[Type[GcpResource]] = [
+    GcpVertexAIModel,
+    GcpVertexAIDataset,
+    # GcpVertexAIDatasetVersion, : collected via GcpVertexAIFeatureGroup
```
Comment (aquamatthias) [inline plugins/gcp/fix_plugin_gcp/resources/aiplatform.py:3359]: can we uncomment them? So we can get the list of resources. 
It is fine if the resources are collected via another resource.

File: plugins/gcp/fix_plugin_gcp/resources/aiplatform.py [hunk 2]
```diff
@@ -0,0 +1,3376 @@
...
+    "southamerica-west1",
+    "us-central1",
+    "us-east1",
+    "us-east4",
+    "us-south1",
+    "us-west1",
+    "us-west2",
+    "us-west3",
+    "us-west4",
+]
```
Comment (aquamatthias) [inline plugins/gcp/fix_plugin_gcp/resources/aiplatform.py:60]: There is no good way to synchronize this information. I would make it more stupid: let's collect in all regions and ignore the errors if not existent.

File: plugins/gcp/fix_plugin_gcp/resources/aiplatform.py (hunk 1 above)
Comment (1101-1) [reply plugins/gcp/fix_plugin_gcp/resources/aiplatform.py]: We use method `collect_resources` on each of commented resources. for example:
```python
@define(eq=False, slots=False)
class GcpVertexAIDataset(AIPlatformRegionFilter, GcpResource):
    @classmethod
    def collect(cls: Type[GcpResource], raw: List[Json], builder: GraphBuilder) -> List[GcpResource]:
        # Additional behavior: iterate over list of collected GcpVertexAIDataset and for each:
        # - collect related GcpVertexAIDatasetVersion
        result: List[GcpResource] = super().collect(raw, builder)  # type: ignore
        dataset_ids = [dataset.id for dataset in cast(List[GcpVertexAIDataset], result)]
        for dataset_id in dataset_ids:
            builder.submit_work(GcpVertexAIDatasetVersion.collect_resources, builder, parent=dataset_id)

        return result
```
If we uncomment these sub-resources, then collector will use their `api_spec` and will not be able to collect the data.

Comment (aquamatthias) [reply plugins/gcp/fix_plugin_gcp/resources/aiplatform.py]: Mhh I am sure that would be solvable - it is not that important to block this PR.

**Review by aquamatthias — APPROVED** [review]

**Pull request merged**